File: amamov/실패율.js
function solution(N, stages) {
    let result = {}
    const stageNums = [];
    
    for (let i=1; i < N+1; i++) {
        stageNums.push(i);
    }
    
    stageNums.forEach(curStage => {
        let loser = 0
        const all = stages.filter(s => {
            if (s === curStage) {
                loser ++;
            }
            return s >= curStage
        }).length;
        const failRatio = loser / all
        result[curStage] = failRatio
    })
    
    result = Object.keys(result).map((key) => {
        return [key, result[key]];
    });
    
    result.sort((first, second) => {
        return second[1] - first[1];
    });
    
    const answer = []
       
    return result.map(r => Number(r[0]));
}

File: amamov/실패율.py
def filter_loser(stages, cur_stage):
    a, b = 0, 0
    for stage in stages:
        if stage >= cur_stage:
            b += 1
        if stage == cur_stage:
            a += 1
    return a, b

def solution(N, stages):
    result = {}
    for cur_stage in range(1, N+1):
        a, b = filter_loser(stages, cur_stage)
        if b == 0:
            result[cur_stage] = 0
        else:
            result[cur_stage] = a / b
    result = sorted(result.items(), key=lambda x: x[1], reverse=True)
    print(result)
    return [int(r[0]) for r in result]
